var/spack/repos/builtin/packages/coreneuron/package.py

```diff
@@ -57,15 +59,17 @@ class Coreneuron(CMakePackage):
 
     depends_on('boost', when='+tests')
     depends_on('cuda', when='+gpu')
-    depends_on('flex', type='build')
+    depends_on('flex', type='build', when='~nmodl')
+    depends_on('flex@2.6:', type='build', when='+nmodl')
     depends_on('mpi', when='+mpi')
     depends_on('reportinglib', when='+report')
     depends_on('libsonata-report', when='+report')
     depends_on('reportinglib+profile', when='+report+profile')
     depends_on('tau', when='+profile')
 
     # nmodl specific dependency
-    depends_on('nmodl@0.3b:', when='@0.17:+nmodl')
+    depends_on('nmodl@0.3:', when='@1.0.0:+nmodl')
```

Comment:
Isn't there the same version issue here?

Reply:
Yes, you're right. As written this accepts 0.3a and 0.3b with coreneuron 1.0.0. I'll change nmodl to be called 0.3.0.

This isn't very nice, but nor are the two obvious alternatives I see:
 - removing the alpha/beta versions
 - hoping that it never **really** matters that spack sorts these versions in the wrong order in dependencies etc.

Reply:
Pretty annoying, IMO.
```
>>> from spack import Version
>>> Version("0.3.0dev0") > Version("0.3.0")
True
>>> Version("0.3.0.dev0") > Version("0.3.0")
True
>>> Version("0.3.0-beta") > Version("0.3.0")
True
>>> Version("0.3.0_2010_13_32") > Version("0.3.0")
True
```

OTOH, I can see how there's no clean way to do this, really. Wasn't there once upon a time the convention to use even releases for stable / uneven for dev? Or at least use `0.2.999b123`?

Reply:
I agree that there doesn't seem to be a really nice option. Given the starting point with CoreNEURON/NMODL, I don't think the proposed change here with an extra `.0` is THAT bad.

For the next releases, maybe we should just adopt this (1.1.0 alpha can be 1.1.a or 1.1a etc.) 🤷